docs/gettinginvolved/infrastructure/release-process.rst

```diff
@@ -312,24 +312,6 @@ Release
       Connection to 127.0.0.1 closed.
       $ vagrant scp default:/home/vagrant/.aws ~/
 
-#. Submit the release pull request for review again.
-
-Post-Release Review Process
```

Comment:
We don't have any QA process remaining if we stop testing these instructions (I think we should) so this section is removed and the merge happens earlier. This will stop us having the constant problem of having to copy and paste already-reviewed release notes and NEWS from one failed release's PR to a new PR.